### ISSUE TITLE
Random suffix in service name

### ISSUE THREAD

**Issue by wstryjew** [Apicurio/apicurio-registry-operator#71]
Apicurio registry service name has random suffix after installation with operator.

When I install apicurio as described in quickstart (https://github.com/Apicurio/apicurio-registry-operator#quickstart), service name has random suffix. Service name should be constant value or a value defined in mainifest. When it has random suffix I can not use k8s dns with fixed service name.

Is it possible to configure the service name via any property or env variable?

Deployed on:
`Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.2", GitCommit:"cff46ab41ff0bb44d8584413b598ad8360ec1def", GitTreeState:"clean", BuildDate:"2019-01-10T23:35:51Z", GoVersion:"go1.11.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.2", GitCommit:"cff46ab41ff0bb44d8584413b598ad8360ec1def", GitTreeState:"clean", BuildDate:"2019-01-10T23:28:14Z", GoVersion:"go1.11.4", Compiler:"gc", Platform:"linux/amd64"}`

**Comment by rmarting** [comment]
It could be a good idea to use the original name of the service-registry (metadata.name property in custom resource) to declare the service name. So if we have an ```ApicurioRegistry``` definition similar to:

```
apiVersion: apicur.io/v1alpha1
kind: ApicurioRegistry
metadata:
  name: service-registry
spec:
  configuration:
    persistence: "streams"
    streams:
      bootstrapServers: "my-kafka-kafka-bootstrap:9092"
  deployment:
    replicas: 1
```

The service name could be declared as ```service-registry-svc``` and the route exposed only the metadata.name. With this approach it will be easy to identify the resources created by our ApicurioRegistry definition.

What do you think?

**Comment by wstryjew** [comment]
I think it is good idea. Service name will be unique and well defined. What's more the name suffixed with "svc" suggests it is a service.

**Comment by gerbrand** [comment]
Is there any update on this? Even environment-names that are defined by operator have the random string added (e.g.)
`declare -x APICURIOREGISTRY_SERVICE_FH4LN_PORT_8080_TCP_ADDR="10.244.91.139"
`
I can't imagine I am the only one puzzled by this. Is there a way to remove that random prefix via configuration? Or can I help with a patch/pr ?

**Comment by jsenko** [comment]
Hi @gerbrand thanks for pointing this out again, no progress on this yet, but a PR would be welcome. 
I like the idea of a service name with a fixed suffix. The related code is here https://github.com/Apicurio/apicurio-registry-operator/blob/0.x/pkg/controller/apicurioregistry/svc/factory/factory_kube.go#L69 and in the other factory implementations, and I think it could be replaced by `Name: this.ctx.GetAppName().Str() + "-" + typeTag`.

**Comment by gerbrand** [comment]
Thanks for the reploy @jsenko  So just removing  the trailing "-" ? I could try out on my local machine, but I wonder why it is there in the first place.

**Comment by jsenko** [comment]
And using `Name` instead of `GenerateName`. It just didn't occur to me that consistent naming would be important.

**Comment by jsenko** [comment]
If you create a PR that would be great, otherwise I'll get back to this a bit later.

**Comment by gerbrand** [comment]
The random postfix is added at a later point. Removing the trailing - only resulted in the postfix not having a -. I could not find anywhere in the code where the random postfix is generated and added. I'm slightly to less experienced in Go to navigate to the indirection. Any clue where that is done? Via Kubernetes-internals?

**Comment by gerbrand** [comment]
As a workaround I defined my own service:

```
---
apiVersion: v1
kind: Service
metadata:
  name: apicurioregistry
spec:
  ports:
    - name: "api"
      port: 8080
      targetPort: 8080
  selector:
    app: apicurioregistry

```

Using above, an extra service is crated with name apicurioregistry without any prefix. The selector is the same as the original service created by the operator. That way, other application/services within the same kubernetes cluster can access the service via the url http://apicurioregistry:8080/api

**Comment by jsenko** [comment]
Hello, we fixed this issue in `1.0.0-dev` (`master`), it should be resolved after the official `1.0.0` release.
